lib/date_time/date_time_modem.c

```diff
@@ -224,3 +226,12 @@ void date_time_modem_xtime_subscribe(void)
 		LOG_ERR("Subscribing to modem AT%%XTIME notifications failed, err=%d", err);
 	}
 }
+
+K_WORK_DEFINE(date_time_modem_xtime_subscribe_work, date_time_modem_xtime_subscribe_work_fn);
+
+void date_time_modem_xtime_subscribe(void)
```

Comment:
We could call this from `NRF_MODEM_LIB_ON_CFUN` callback instead of when `CEREG: 2` is received.

Reply:
Yes, most likely that was the intention of the current implementation, but the callback wasn't available at the time. CFUN callback is a much better way to do it.

Reply:
This is now done